mesonbuild/coredata.py

```diff
@@ -718,6 +718,13 @@ def set_options(self, options, *, subproject='', warn_unknown=True):
             self.copy_build_options_from_regular_ones()
 
     def set_default_options(self, default_options, subproject, env):
+        # Warn if the user is using two different ways of setting build-type
+        # options that override each other
+        if 'buildtype' in env.cmd_line_options and \
+           ('optimization' in env.cmd_line_options or 'debug' in env.cmd_line_options):
+            mlog.warning('Recommend using either -Dbuildtype or -Doptimization + -Ddebug. '
+                         'Using both is redundant since they override each other. '
+                         'See: https://mesonbuild.com/Builtin-options.html#build-type-options')
```

Comment:
Would be nice to add a title above the buildtype/debug/optimization table so we can link directly to that part of the doc.

Reply:
Good point.

Reply:
Wow, I just realized that I thought I already did that. I added the anchor in the URL but didn't add it to the documentation.